### DISCUSSION TITLE
Search not working with Roundcube

### DISCUSSION THREAD

**Discussion by vitobotta** [Mailu/Mailu#3078] [User support] [answered]
Hi! first of all thanks for this awesome project! It was super easy to set up and also backing up is ridiculously easy.

I migrated a few domains and mailboxes to a new Mailu install yesterday, and the migration completed over a day ago but the indexer-worker process is still using a lot of CPU. Could it really still be indexing my emails? I think it's around 90-100K messages. Also when I try searching for a message using Roundcube, it doesn't find anything basically. How to troubleshoot? Thanks!

**Comment by nextgens** [comment]
Yes, odds are the indexing has not finished.

Which version of Mailu is that with? We have made major changes to how indexing/FTS works in master

**Reply by vitobotta** [reply]
Once this process is finished, will search be quick when using the webmail? I installed Mailu a couple of days ago so it's whichever version that gets installed by following the docs. The Docker images have tag 2.0. Shall I switch to master? (if yes, how?)

**Reply by nextgens** [reply]
Yes, search should be fast.

Not necessarily, using the stable version is fine. Master brings in new features (non-english support for the stemmer, indexing of attachments, new backend engine, ...).

Try forcing the indexation:
```
docker compose exec imap doveadm index -A '*'
```
This should finish reasonably quickly if the indexes are up to date/ready.

**Reply by vitobotta** [reply]
Will try that now, thanks!

**Reply by vitobotta** [reply]
The command you gave me is still running but showed this "Warning: FTS Xapian: Memory too low (text) 'std::bad_alloc'". However there's almost 5GB of RAM free on the server. What can I do?

**Reply by vitobotta** [reply]
Could it be due to the setting "LD_PRELOAD=/usr/lib/libhardened_malloc.so"?

**Reply by nextgens** [reply]
Hmm this is because all the indexing happens in parallel. It's fixed on master...

Try this instead:
```
docker compose exec imap doveadm user '*'|while read line; do echo "$line"; docker compose exec imap doveadm index -u "$line" '*'; done
```

**Reply by vitobotta** [reply]
How do I stop the previous command? Ctrl-c is not working

**Reply by vitobotta** [reply]
I killed the previous and the new one tells me "the input device is not a TTY". Thanks for helping btw!

**Reply by vitobotta** [reply]
It's working with a slightly different command "docker compose exec imap bash -c "doveadm user '*' | while read line; do echo \$line; doveadm index -u \$line '*'; done"" - let's see how it goes :)

**Reply by vitobotta** [reply]
It works for a while and then I get again "Warning: FTS Xapian: Memory too low (text) 'std::bad_alloc'"

**Reply by nextgens** [reply, marked as answer]
I am not sure what to tell you here.

Options:
- add more RAM
- turn off ``FULL_TEXT_SEARCH`` (and stay on stable)
- upgrade to master (we do not have a good documentation on what to do there yet)

If you do upgrade, ensure you at least:
- re-run setup (and consider enabling Tika if you want attachments indexed too); You will need an updated docker-compose.yml
- update ``FULL_TEXT_SEARCH`` (now it takes a comma separated list of languages; see the doc)
- re-index (and maybe remove existing xapian indexes)

**Reply by vitobotta** [reply]
The weird thing is that there is a lot of free memory in the system while this is running. Is there any memory limit defined 
 somewhere in the stack?

**Reply by nextgens** [reply]
Not that I know of.

dovecot-fts-xapian is buggy (indexes get corrupted and keep growing), this is why we are switching to dovecot-fts-flatcurve

**Reply by vitobotta** [reply]
Has this changed in master already?

**Reply by vitobotta** [reply]
I would like to switch to master if there are improvements in this area and more. Are there any risks at the moment?

**Reply by nextgens** [reply]
We are planning to make a release within a month.

To switch, re-run setup and pick "master"

**Reply by vitobotta** [reply]
Gotcha, will try that, thanks!

**Reply by vitobotta** [reply]
Upgraded! How do I reindex with this version?

**Reply by vitobotta** [reply]
I am monitoring the stats command and it seems it's indexing already! 6500 emails so far. Looking better.

**Reply by vitobotta** [reply]
@nextgens indexing has completed and the search via Roundcube is super fast now! :D Thanks for your help! Should I do anything to remove/clean up the old index data with xapian or something?

**Reply by nextgens** [reply]
That's the part of the instructions that is not ready.

I believe you have a xapian-index folder in the - maildirs you can safely dispose of.

**Reply by vitobotta** [reply]
Perfect, I have removed them. Thanks again!
